modules/scylladb/examples_test.go

```diff
@@ -0,0 +1,226 @@
+
+	connectionHost, err := scyllaContainer.ConnectionHost(ctx, 8080) // Alternator port
+	if err != nil {
+		log.Printf("failed to get connection host: %s", err)
+		return
+	}
+
+	runGoCQLExampleTest(connectionHost)
+
+	// Output:
+	// true
+}
+
+func ExampleRun_withShardAwareness() {
+	ctx := context.Background()
+
+	// runScyllaDBContainerWithShardAwareness {
+	scyllaContainer, err := scylladb.Run(ctx,
+		"scylladb/scylla:6.2",
+		scylladb.WithShardAwareness(),
+	)
+	// }
+	defer func() {
+		if err := testcontainers.TerminateContainer(scyllaContainer); err != nil {
+			log.Printf("failed to terminate container: %s", err)
+		}
+	}()
+	if err != nil {
+		log.Printf("failed to start container: %s", err)
+		return
+	}
+
+	state, err := scyllaContainer.State(ctx)
+	if err != nil {
+		log.Printf("failed to get container state: %s", err)
+		return
+	}
+
+	fmt.Println(state.Running)
+
+	connectionHost, err := scyllaContainer.ConnectionHost(ctx, 19042) // ShardAwareness port
+	if err != nil {
+		log.Printf("failed to get connection host: %s", err)
+		return
+	}
+
+	runGoCQLExampleTest(connectionHost)
+
+	// Output:
+	// true
+}
+
+func ExampleRun_withConfigFile() {
+	ctx := context.Background()
+
+	// runScyllaDBContainerWithConfigFile {
+	scyllaContainer, err := scylladb.Run(ctx,
+		"scylladb/scylla:6.2",
+		scylladb.WithConfigFile(filepath.Join("testdata", "scylla.yaml")),
+	)
+	// }
+	defer func() {
+		if err := testcontainers.TerminateContainer(scyllaContainer); err != nil {
+			log.Printf("failed to terminate container: %s", err)
+		}
+	}()
+	if err != nil {
+		log.Printf("failed to start container: %s", err)
+		return
+	}
+	// }
+
+	state, err := scyllaContainer.State(ctx)
+	if err != nil {
+		log.Printf("failed to get container state: %s", err)
+		return
+	}
+
+	fmt.Println(state.Running)
+
+	connectionHost, err := scyllaContainer.ConnectionHost(ctx, 9042) // Non ShardAwareness port
+	if err != nil {
+		log.Printf("failed to get connection host: %s", err)
+		return
+	}
+
+	runGoCQLExampleTest(connectionHost)
+
+	// Output:
+	// true
+}
+
+func ExampleRun() {
+	// runBaseScyllaDBContainer {
+	ctx := context.Background()
+
+	scyllaContainer, err := scylladb.Run(ctx,
+		"scylladb/scylla:6.2",
+		scylladb.WithShardAwareness(),
+	)
+	defer func() {
+		if err := testcontainers.TerminateContainer(scyllaContainer); err != nil {
+			log.Printf("failed to terminate container: %s", err)
+		}
+	}()
+	if err != nil {
+		log.Printf("failed to start container: %s", err)
+		return
+	}
+	// }
+
+	state, err := scyllaContainer.State(ctx)
+	if err != nil {
+		log.Printf("failed to get container state: %s", err)
+		return
+	}
+
+	fmt.Println(state.Running)
+
+	// BaseConnectionHost {
+	connectionHost, err := scyllaContainer.ConnectionHost(ctx, 9042)
+	// }
+	if err != nil {
+		log.Printf("failed to get connection host: %s", err)
+		return
+	}
+	runGoCQLExampleTest(connectionHost)
+
+	// Output:
+	// true
+}
+
+func runGoCQLExampleTest(connectionHost string) {
```

Comment:
suggestion: let's make this function to return the error, and fail the examples calling it if non nil.